src/LasagnaController.cpp

```diff
@@ -32,23 +55,42 @@ bool LasagnaController::update(Input input){
     case EQUIL:
       break;
   }
+
   state.v_dldt = input.op*constants.v_dldt_a + constants.v_dldt_b; 
-  state.v1 = v1_filter.update(input.h_rel);
-  state.v2 = v2_filter.update(input.h_rel);
-  state.v = (state.status==EQUIL) ? state.v1 : state.v2;
-  float dldt_total = 0;
+  state.v = (state.status==EQUIL) ? state.v1 : state.v2; // v1 is low corner freqency, so it's used after equilibration
+
+  /**
+   * Predict changes in velocity from external dldt and dldt from actions. This is only done after the
+   * balloon has equilibrated. We then sum all these delta v's (state.dv_sum). We add this sum to the 
+   * fused_v, however we also substract the dv_sum passed through the action_filter. This is becuase the 
```

Comment:
Big fan of this comment. Minor grammatical things that you can ignore: it should read `fused_v; however we also subtract the dv_sum passed through the action_filter. This is because the `

Reply:
lol tbh that's very mild compared to my usual glaring grammer problems